### PR TITLE
Feature/json request

### LINKED ISSUE #56: Feature request: JSON Body
Hi!

I don't know if this is implemented or not, but it would be super useful if we would send the HTTP request body as JSON. Today, only Protobuf is an alternative for sending requests and JSON is a good alternative for sending request as there are short and simple.

Thanks,

### PR DESCRIPTION
**Pull request by kolmant**
This pull requests resolves #56 by adding a Radio button + TextArea for setting a JSON body there. Also, it adds an automatic `Content-Type` based on the selected request body type as follows:

- None: no `Content-Type` header added.
- JSON: `Content-Type: application/json` header added.
- Protobuf: `Content-Type: application/x-protobuf` header added.

Basic usage:

Screen Shot 2021-07-01 at 1 07 25 PM
Screen Shot 2021-07-01 at 1 07 31 PM

Let me know if you need a video demonstration or something else (it's my first time doing something in React / Electron)

### CODE REVIEW

**Review by lbriceno92 — APPROVED** [review]
looks good

File: src/renderer/models/request_builder.ts [hunk 1]
```diff
@@ -32,17 +34,37 @@ export async function toRequestDescriptor(
...
 
   let body;
+  let contentType;
+
   if (bodyType === 'protobuf' && bodies.protobuf) {
     const withEnv = applyToProtoMessage(bodies.protobuf, (s: string): string => applyEnvs(s, varMap));
     body = await serializeProtobuf(withEnv, ctx);
+    contentType = 'application/x-protobuf';
+  } else if (bodyType === 'json') {
+    const withEnv = bodies.json || null;
```
Comment (kolmant) [inline src/renderer/models/request_builder.ts:44]: @pandawithcat do you think it's worth to invoke `applyEnvs` with the JSON body?

**Review by apuli — APPROVED** [review]
Very helpful

**Comment by emilioq48** [comment]
would love to see that feature, totally an upgrade : D

**Review by emilioq48 — APPROVED** [review]

**Comment by DonghunLouisLee** [comment]
> This pull requests resolves #56 by adding a Radio button + TextArea for setting a JSON body there. Also, it adds an automatic `Content-Type` based on the selected request body type as follows:
> 
> * None: no `Content-Type` header added.
> * JSON: `Content-Type: application/json` header added.
> * Protobuf: `Content-Type: application/x-protobuf` header added.
> 
> Basic usage:
> 
> Screen Shot 2021-07-01 at 1 07 25 PM
> 
> Screen Shot 2021-07-01 at 1 07 31 PM
> 
> Let me know if you need a video demonstration or something else (it's my first time doing something in React / Electron)

Thanks for your PR. I'm open to sending json as the http body but I would recommend using Postman for JSON requests primarily because Postman has bette json support. Having said that, I do understand that some people are using both JSON and protobuf for some HTTP requests so I'll merge this PR for now.

**Comment by kolmant** [comment]
> > This pull requests resolves #56 by adding a Radio button + TextArea for setting a JSON body there. Also, it adds an automatic `Content-Type` based on the selected request body type as follows:
> > 
> > * None: no `Content-Type` header added.
> > * JSON: `Content-Type: application/json` header added.
> > * Protobuf: `Content-Type: application/x-protobuf` header added.
> > 
> > Basic usage:
> > Screen Shot 2021-07-01 at 1 07 25 PM
> > Screen Shot 2021-07-01 at 1 07 31 PM
> > Let me know if you need a video demonstration or something else (it's my first time doing something in React / Electron)
> 
> Thanks for your PR. I'm open to sending json as the http body but I would recommend using Postman for JSON requests primarily because Postman has bette json support. Having said that, I do understand that some people are using both JSON and protobuf for some HTTP requests so I'll merge this PR for now.

Thanks for your comment! I totally agree with you! This solves the use case of JSON as request and Protobuf as response in HTTP. For non-mixed cases as JSON-JSON I suggest using Postman too.

**Pull request merged**